torch/distributions.py

```diff
@@ -77,6 +78,20 @@ def log_prob(self, value):
         """
         raise NotImplementedError
 
+    def enumerate_support(self):
+        """
+        Returns tensor containing all values supported by a discrete
+        distribution. The result will enumerate over dimension 0.
+
+        Note that this enumerates over all batched variables in lock-step
+        ``[[0, 0], [1, 1], ...]``. To iterate over the full Cartesian product
```

Comment:
Is this always equivalent to taking a flat support vector, and expanding it to have as many columns as batch size?

Reply:
Correct. The final shape should be `(cardinality,) + batch_shape`. In multivariate distributions like Multinomial or OneHotCategorical, the shape will be `(cardinality,) + batch_shape + event_shape`, but for the simple distributions in this PR, `event_shape = ()`.

Reply:
Done, thanks for the suggestion.